solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -255,6 +263,34 @@ contract LiquidityProtection is TokenHandler, Utils, Owned, ReentrancyGuard, Tim
         store.withdrawTokens(liquidity1.poolToken, msg.sender, liquidity1.poolAmount.add(liquidity2.poolAmount));
     }
 
+    /**
+     * @dev adds protected liquidity to a pool for a specific recipient
+     * also mints new governance tokens for the caller if the caller adds network tokens
+     *
+     * @param _recipient       protected liquidity recipient
+     * @param _poolAnchor      anchor of the pool
+     * @param _reserveToken    reserve token to add to the pool
+     * @param _amount          amount of tokens to add to the pool
+     * @return new protected liquidity id
+     */
+    function addLiquidityFor(
+        address _recipient,
+        IConverterAnchor _poolAnchor,
+        IERC20Token _reserveToken,
+        uint256 _amount
+    )
+        external
+        payable
+        protected
+        validAddress(_recipient)
+        poolSupported(_poolAnchor)
+        poolWhitelisted(_poolAnchor)
+        greaterThanZero(_amount)
+        returns (uint256)
+    {
+        return addLiquidity(_recipient, _poolAnchor, _reserveToken, _amount);
```

Comment:
any reason not to have the addLiquidity logic here?

Reply:
I wanted to avoid double input verification here.

Reply:
oh right, makes sense